### PR TITLE
Dev-6247 Add Search Input

### PR DESCRIPTION
**Pull request by jonhill13**
### Author
- [x] Story Book has been rebuilt (new build artifacts in `/docs`)
- [x] Components have been rebuilt (new build artifacts in `/dist`)

👆These can be done by executing `npm run build` and committing the build artifacts.

- [x] All new components have been exported in `/index.js` `(if applicable)`
- [x] Design Review Complete
- [x] Relevant Draft release notes have been updated describing these changes ([use this template](https://github.com/fedspendingtransparency/data-transparency-ui/blob/master/release_notes_template.md) to document your changes) `(if applicable)`
- [x] Integration Status Updated on relevant documentation page `(if applicable)`
- [x] `index.d.ts` updated with new prop types defined for new components. `(if applicable)`

### Reviewer
- [ ] Code review

### CODE REVIEW

File: components/SearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const [value, setValue] = useState('');
+    const [searchTerm, setSearchTerm] = useState('');
+
+    const resetSearch = () => {
+        setValue('');
+        setQuery('');
+        setSearchTerm('');
+    };
+
+    const onChange = throttle((e) => (changeSearchTerm(searchTerm, e) ? setValue(e.target.value) : resetSearch()), 500);
```
Comment (MaxwellKendall) [inline components/SearchBar.jsx:35]: This seems like a long time to throttle. Can we make this a param?

File: components/SearchBar.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+// the minimum number of characters a user is required to enter before they can perform a search
+const defaultProps = {
+    setQuery: () => {},
+    minChars: 2
+};
+
+const SearchBar = ({ setQuery, minChars }) => {
+    const [value, setValue] = useState('');
+    const [searchTerm, setSearchTerm] = useState('');
+
```
Comment (MaxwellKendall) [inline components/SearchBar.jsx:28]: setQuery, setValue, and setSearchTerm all seem to coney the same thing. Can we add some comments to elaborate?

File: helpers/searchBarHelper.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  if (value && searchTerm === value) {
+      return false;
+  }
+  else if (searchTerm && value.length < minChars) {
+      return false
+  }
+  return true;
+};
+
+export const changeSearchTerm = (searchTerm, e) => {
```
Comment (MaxwellKendall) [inline helpers/searchBarHelper.js:11]: by convention i think e should be the first param

File: helpers/searchBarHelper.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export const resetOrSubmitSearch = (value, searchTerm, minChars) => {
```
Comment (MaxwellKendall) [inline helpers/searchBarHelper.js:1]: value and searchTerm seem synonomous. What's the difference? Can we add comments or change the name of these params.

**Review by MaxwellKendall — CHANGES_REQUESTED** [review]

File: components/SearchBar.jsx (hunk 1 above)
Comment (jonhill13) [reply components/SearchBar.jsx:35]: Yeah, that is a good call.

**Review by MaxwellKendall — APPROVED** [review]

**Pull request merged**